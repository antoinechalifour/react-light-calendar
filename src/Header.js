import React, { Component } from 'react'

class Header extends Component {
  changeMonth (numberOfMonthToAdd) {
    const { currentMonth, onChange } = this.props

    return () => {
      const nextMonth = currentMonth.clone()
      nextMonth.add(numberOfMonthToAdd, 'month')

      onChange(nextMonth)
    }
  }

  render () {
    const {
      currentMonth,
      style = {}
    } = this.props
    const { headerStyle = {} } = style
    const arrowBaseStyle = {
      position: 'absolute',
      top: '50%',
      transform: 'translateY(-50%)',
      cursor: 'pointer'
    }

    return (
      <div
        style={Object.assign({
            // Default style
          textTransform: 'uppercase',
          borderBottom: '1px solid #d7d8d9',
          textAlign: 'center',
          padding: '12px'
        },
        // Custom user-defined style
        headerStyle, {
          // Required styles
          position: 'relative'
        })}
      >
        <div
          style={Object.assign({},
            // Arrow default style
            arrowBaseStyle, {
              left: '8px'
            }
          )}
          onClick={this.changeMonth(-1)}
        >
          {'<'}
        </div>

        <div>{currentMonth.format('MMMM YYYY')}</div>

        <div
          style={Object.assign({},
            // Arrow default style
            arrowBaseStyle, {
              right: '8px'
            }
          )}
          onClick={this.changeMonth(1)}
        >
          {'>'}
        </div>
      </div>
    )
  }
}

Header.propTypes = {
  currentMonth: React.PropTypes.object.isRequired,
  onChange: React.PropTypes.func.isRequired,
  style: React.PropTypes.object
}

export default Header
