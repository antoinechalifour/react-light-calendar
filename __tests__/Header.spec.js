import React from 'react'
import ReactTestRenderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Header from '../src/Header'
import moment from 'moment'

it('Should render the month with the year', () => {
  moment.locale('en')
  const currentMonth = moment('01/02/2017', 'DD/MM/YYYY')
  const renderer = ReactTestRenderer.create(
    <Header
      currentMonth={currentMonth}
      onChange={function onChange () {}}
    />
  )

  expect(renderer.toJSON()).toMatchSnapshot()
})

it('Should display the month according to the locale', () => {
  moment.locale('fr')
  const currentMonth = moment('01/02/2017', 'DD/MM/YYYY')
  const renderer = ReactTestRenderer.create(
    <Header
      currentMonth={currentMonth}
      onChange={function onChange () {}}
    />
  )

  expect(renderer.toJSON()).toMatchSnapshot()
})

it('Should trigger onChange(prevMonth) on previous click', () => {
  moment.locale('en')
  const currentMonth = moment('01/02/2017', 'DD/MM/YYYY')
  const expected = moment('01/01/2017', 'DD/MM/YYYY')
  const onChange = jest.fn()
  const wrapper = shallow(
    <Header
      currentMonth={currentMonth}
      onChange={onChange}
    />
  )

  const prev = wrapper.find('div > div').at(0)
  prev.simulate('click')
  const actual = onChange.mock.calls[0][0]

  expect(expected.isSame(actual)).toBe(true)
})

it('Should trigger onChange(nextMonth) on next click', () => {
  moment.locale('en')
  const currentMonth = moment('01/02/2017', 'DD/MM/YYYY')
  const expected = moment('01/03/2017', 'DD/MM/YYYY')
  const onChange = jest.fn()
  const wrapper = shallow(
    <Header
      currentMonth={currentMonth}
      onChange={onChange}
    />
  )

  const prev = wrapper.find('div > div').at(2)
  prev.simulate('click')
  const actual = onChange.mock.calls[0][0]

  expect(expected.isSame(actual)).toBe(true)

})

it('Should accept custom styling', () => {
  moment.locale('en')
  const currentMonth = moment('01/02/2017', 'DD/MM/YYYY')
  const renderer = ReactTestRenderer.create(
    <Header
      currentMonth={currentMonth}
      onChange={function onChange () {}}
      style={{
        headerStyle: {
          textTransform: 'none',
          backgroundColor: 'red',
          borderBottom: 'none',
          textAlign: 'right',
          padding: '0',
          margin: 'auto',
          position: 'fixed' // should be skipped
        }
      }}
    />
  )

  expect(renderer.toJSON()).toMatchSnapshot()
})
