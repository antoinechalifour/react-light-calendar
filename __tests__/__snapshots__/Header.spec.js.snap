exports[`test Should accept custom styling 1`] = `
<div
  style={
    Object {
      "backgroundColor": "red",
      "borderBottom": "none",
      "margin": "auto",
      "padding": "0",
      "position": "relative",
      "textAlign": "right",
      "textTransform": "none",
    }
  }>
  <div
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "left": "8px",
        "position": "absolute",
        "top": "50%",
        "transform": "translateY(-50%)",
      }
    }>
    <
  </div>
  <div>
    February 2017
  </div>
  <div
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "position": "absolute",
        "right": "8px",
        "top": "50%",
        "transform": "translateY(-50%)",
      }
    }>
    >
  </div>
</div>
`;

exports[`test Should display the month according to the locale 1`] = `
<div
  style={
    Object {
      "borderBottom": "1px solid #d7d8d9",
      "padding": "12px",
      "position": "relative",
      "textAlign": "center",
      "textTransform": "uppercase",
    }
  }>
  <div
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "left": "8px",
        "position": "absolute",
        "top": "50%",
        "transform": "translateY(-50%)",
      }
    }>
    <
  </div>
  <div>
    f√©vrier 2017
  </div>
  <div
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "position": "absolute",
        "right": "8px",
        "top": "50%",
        "transform": "translateY(-50%)",
      }
    }>
    >
  </div>
</div>
`;

exports[`test Should render the month with the year 1`] = `
<div
  style={
    Object {
      "borderBottom": "1px solid #d7d8d9",
      "padding": "12px",
      "position": "relative",
      "textAlign": "center",
      "textTransform": "uppercase",
    }
  }>
  <div
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "left": "8px",
        "position": "absolute",
        "top": "50%",
        "transform": "translateY(-50%)",
      }
    }>
    <
  </div>
  <div>
    February 2017
  </div>
  <div
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "position": "absolute",
        "right": "8px",
        "top": "50%",
        "transform": "translateY(-50%)",
      }
    }>
    >
  </div>
</div>
`;
